datastore/kubernetes.go

```diff
@@ -423,6 +423,16 @@ func (s *DataStore) GetKubernetesNode(name string) (*corev1.Node, error) {
 	return s.knLister.Get(name)
 }
 
+// IsKubeNodeUnschedulable checks if the Kubernetes Node resource is
+// unschedulable
+func (s *DataStore) IsKubeNodeUnschedulable(nodeName string) (bool, error) {
```

Comment:
I used the IsNodeSchedulable from the lh node (which is synced by the manager), in the share-manager not sure what is better, probably using the kube node status directly is better.

Reply:
Here we should check the kube node status.

For lh nodes, scheduling means replica allocation.
For kube nodes, scheduling means pod allocation.